packages/controllers/src/services/WebWorkerExecutionEnvironmentService.ts

```diff
@@ -134,30 +142,52 @@ export class WorkerController extends SafeEventEmitter {
     console.log(`worker:${workerId} terminated`);
   }
 
-  async startPlugin(
-    workerId: string,
-    pluginData: PluginData,
-  ): Promise<unknown> {
-    const _workerId: string = workerId || this.workers.keys().next()?.value();
-    if (!_workerId) {
-      throw new Error('No workers available.');
-    }
+  /**
+   * Gets the RPC message handler for the given plugin.
+   *
+   * @param pluginName - The name of the plugin whose message handler to get.
+   */
+  async getRpcMessageHandler(pluginName: string) {
+    return this._pluginRpcHooks.get(pluginName);
+  }
 
-    this._mapPluginAndWorker(pluginData.pluginName, workerId);
+  private _removePluginHooks(pluginName: string) {
+    this._pluginRpcHooks.delete(pluginName);
+  }
 
-    return this.command(_workerId, {
+  private _createPluginHooks(pluginName: string, workerId: string) {
+    const rpcHook = async (
+      origin: string,
+      request: Record<string, unknown>,
+    ) => {
+      return await this._command(workerId, {
+        id: nanoid(),
+        jsonrpc: '2.0',
+        method: 'pluginRpc',
+        params: {
+          origin,
+          request,
+          target: pluginName,
+        },
+      });
+    };
+
+    this._pluginRpcHooks.set(pluginName, rpcHook);
+  }
+
+  async executePlugin(pluginData: PluginData): Promise<unknown> {
+    const _workerId = await this._initWorker();
+
+    this._mapPluginAndWorker(pluginData.pluginName, _workerId);
```

Comment:
Ah! We should make sure that it's impossible to execute a plugin twice, by checking if it's already mapped at the top of this function. This is probably worthy of a test case, too.

```suggestion
  async executePlugin(pluginData: PluginData): Promise<unknown> {
    if (this.pluginToWorkerMap.has(pluginData.pluginName)) {
      throw new Error(`Plugin "${pluginData.pluginName}" is already being executed.`);
    }

    const _workerId = await this._initWorker();
    this._mapPluginAndWorker(pluginData.pluginName, _workerId);
```